docker-environment-driver/src/test/java/org/jboss/pnc/environment/docker/DockerEnvironmentDriverRemoteTest.java

```diff
@@ -0,0 +1,245 @@
+        // Create container
+        Environment environment = new Environment(BuildType.DOCKER, OperationalSystem.LINUX);
+        DockerRunningEnvironment runningEnv = (DockerRunningEnvironment)
+                dockerEnvDriver.buildEnvironment(environment, APROX_DEPENDENCY_URL, APROX_DEPLOY_URL);
+
+        try {
+            testRunningContainer(runningEnv, true, "Environment wasn't successfully built.");
+        } catch (Exception | AssertionError e) {
+            dockerEnvDriver.destroyEnvironment(runningEnv.getId());
+            throw e;
+        }
+
+        // Destroy container
+        dockerEnvDriver.destroyEnvironment(runningEnv.getId());
+        testRunningContainer(runningEnv, false, "Environment wasn't successfully destroyed.");
+    }
+
+    @Test
+    public void copyFileToContainerStringDataTest() throws Exception {
+        copyFileToContainerInvariantData("TEST CONTENT", null);
+    }
+
+    @Test
+    public void copyFileToContainerStreamDataTest() throws Exception {
+        copyFileToContainerInvariantData(null, new ByteArrayInputStream("TEST CONTENT".getBytes("UTF-8")));
+    }
+
+    private void copyFileToContainerInvariantData(String string, InputStream stream) throws Exception {
+        Environment environment = new Environment(BuildType.DOCKER, OperationalSystem.LINUX);
+
+        DockerRunningEnvironment runningEnv = (DockerRunningEnvironment)
+                dockerEnvDriver.buildEnvironment(environment, APROX_DEPENDENCY_URL, APROX_DEPLOY_URL);
+
+        String pathToFile = "/tmp/testFile-" + UUID.randomUUID().toString() + ".txt";
+
+        // Get content of container's file system
+        String fsContentOld = processGetRequest(dockerControlEndpoint + "/containers/"
+                + runningEnv.getId() + "/changes");
+        fsContentOld.contains(pathToFile);
+
+        try {
+            dockerEnvDriver.copyFileToContainer(runningEnv.getSshPort(), pathToFile, string, stream);
+        } catch (Exception | AssertionError e) {
+            dockerEnvDriver.destroyEnvironment(runningEnv.getId());
+            throw e;
+        }
+
+        // Get content of container's file system
+        String fsContentNew = processGetRequest(dockerControlEndpoint + "/containers/"
+                + runningEnv.getId() + "/changes");
+        assertTrue("File was not coppied to the container.", fsContentNew.contains(pathToFile));
+
+        dockerEnvDriver.destroyEnvironment(runningEnv.getId());
+    }
+
+    /**
+     * Checks if container was started and the services are on.
+     * 
+     * @param runningEnv Connection data about environment to test
+     * @param shouldBeRunning Indicates, if the environment should be running or should be not available
+     * @param baseErrorMsg Prefix of error message, which is printed if the container is not in expected state
+     */
+    private void testRunningContainer(DockerRunningEnvironment runningEnv, boolean shouldBeRunning,
```

Comment:
Maybe using custom assertions would be a better idea (maybe not now, but in the future?): they look much nicer: http://joel-costigliola.github.io/assertj/assertj-core-custom-assertions.html